File: Fibonacci in the array/Fibonacci_in_the_array.py
class Solution:
    def fibonacci(self, st, maxi):
        first, second = 0, 1
        st.add(first)
        st.add(second)
        while first + second <= maxi:
            temp = first + second
            st.add(temp)
            first, second = second, temp
            
            
    def countFibonacciNumbers(self, arr):
        maxi = max(arr)
        st = set()
        self.fibonacci(st, maxi)
        cnt = 0
        for i in arr:
            if i in st:
                cnt += 1
        return cnt

File: Fibonacci in the array/Fibonacci_in_the_array.java
class Solution {
    public void fibonacci(Set<Long> fibSet, long maxVal) {
        long first = 0, second = 1;
        fibSet.add(first);
        fibSet.add(second);
        while (first + second <= maxVal) {
            long temp = first + second;
            fibSet.add(temp);
            first = second;
            second = temp;
        }
    }
    public int countFibonacciNumbers(long[] arr) {
        long maxVal = Arrays.stream(arr).max().getAsLong();
        Set<Long> fibSet = new HashSet<>();
        fibonacci(fibSet, maxVal);
        int count = 0;
        for (long num : arr) {
            if (fibSet.contains(num)) {
                count++;
            }
        }
        return count;
    }
}